numpy/distutils/fcompiler/gnu.py

```diff
@@ -122,26 +122,18 @@ def get_flags_linker_so(self):
             # error checking.
             if not target:
                 # If MACOSX_DEPLOYMENT_TARGET is not set in the environment,
-                # we try to get it first from the Python Makefile and then we
-                # fall back to setting it to 10.3 to maximize the set of
+                # we try to get it first from sysconfig and then
+                # fall back to setting it to 10.7 to maximize the set of
                 # versions we can work with.  This is a reasonable default
                 # even when using the official Python dist and those derived
                 # from it.
-                import distutils.sysconfig as sc
-                g = {}
-                try:
-                    get_makefile_filename = sc.get_makefile_filename
-                except AttributeError:
-                    pass  # i.e. PyPy
-                else:
-                    filename = get_makefile_filename()
-                    sc.parse_makefile(filename, g)
-                target = g.get('MACOSX_DEPLOYMENT_TARGET', '10.3')
-                os.environ['MACOSX_DEPLOYMENT_TARGET'] = target
-                if target == '10.3':
-                    s = 'Env. variable MACOSX_DEPLOYMENT_TARGET set to 10.3'
+                import sysconfig
+                target = sysconfig.get_config_var('MACOSX_DEPLOYMENT_TARGET')
```

Comment:
What's the difference between `distutils.sysconfig.get_config_var` and `sysconfig.get_config_var`?

Reply:
`sysconfig` is new from python 2.7. The [release note](https://docs.python.org/2/whatsnew/2.7.html#new-module-sysconfig) mentions the attempt to write a distutils2 package. At the time I think Python wanted to deprecate distutils, so they moved this module out. Since the split, `sysconfig` continues to be developed while `distutils.sysconfig` I think is basically frozen.